applications/tari_console_wallet/src/main.rs

```diff
@@ -108,6 +114,19 @@ fn main_inner() -> Result<(), ExitCodes> {
         return runtime.block_on(change_password(&global_config, arg_password, shutdown_signal));
     }
 
+    // Run our own Tor instance, if configured
+    if global_config.console_wallet_use_libtor &&
+        matches!(global_config.comms_transport, CommsTransport::TorHiddenService { .. })
+    {
+        let tor = Tor::randomize(19_050..20_000)?;
```

Comment:
The tor port will not conflict with TCP ports (they are completely different protocols, tor port doesn't bind on a network socket) so the range can be wider - but otherwise, no problem keeping it in this range, though I may simplify this by removing it as an argument because basically the tor port doesn't matter.

 If I understand, this will change the address each time the node starts up. This may cause issues in connectivity because some nodes may not have up to date addresses (once #3629 goes in, it can update more quickly, but still eventual consistency, larger the network longer it may take for some nodes). It's better to change addresses fairly rarely (I would say just keep the address the same).

Reply:
No this is just the control and socks ports the tor identity is still used. Will just confirm that and make it more obvious